### PR TITLE
github workflow: enable workflow approval

### LINKED ISSUE #631: Enable GitHub workflows for PRs with the ok-to-test label
Allow to run GitHub workflows for Pull Requests with the `ok-to-test` label using the prow bot command /ok-to-test. This was already done in the etcd repo (refer to the issue https://github.com/etcd-io/etcd/issues/16956 and its PR https://github.com/etcd-io/etcd/pull/16965).

### PR DESCRIPTION
**Pull request by ivanvc**
Allow to run GitHub workflows for Pull Requests that have the ok-to-test label.

Fixes #631

### CODE REVIEW

**Review by jmhbnz — APPROVED** [review]
LGTM - Great to have more autonomy for etcd-io org members to be enable to trigger and retry workflows, thanks @ivanvc.

Edit: Rather ironically this pull request has 4 workflows that need maintainer approval 😆

File: .github/workflows/gh-workflow-approve.yaml
Comment (ahrtr) [inline .github/workflows/gh-workflow-approve.yaml:10]: Note: there is no `main` branch in bbolt, instead it's `master`.

Comment (ivanvc) [reply .github/workflows/gh-workflow-approve.yaml:10]: Duh, you're right.

Comment (jmhbnz) [reply .github/workflows/gh-workflow-approve.yaml]: Perhaps in future we should align the branch to main @ahrtr?  I believe the procedure is pretty straightforward in GitHub.  It makes sense to try and use same across all sig-etcd subprojects if possible I think.

Comment (ivanvc) [reply .github/workflows/gh-workflow-approve.yaml]: I can leave both `main` and `master` in the meantime, let me know what you think. So, when we rename the branch to `main` there's not gonna be a dependency here.

Comment (ahrtr) [reply .github/workflows/gh-workflow-approve.yaml]: > Perhaps in future we should align the branch to main

OK, let me do it on weekend.

Comment (ahrtr) [reply .github/workflows/gh-workflow-approve.yaml]: cc @jmhbnz @serathius @wenjiaswe @fuweid @tjungblu @Elbehery 

If there is no any objection, I will rename the `master` to `main` on weekend. 

Reference: https://www.git-tower.com/learn/git/faq/git-rename-master-to-main

Comment (ivanvc) [reply .github/workflows/gh-workflow-approve.yaml]: You can also do it easily from the GitHub repository settings page (ref: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch)

Screenshot from 2023-12-08 12-09-56(1)
Screenshot from 2023-12-08 12-10-11


If it's fine with you, I'll revert my last change and set the branch name to `main`, and you can merge after you have renamed the main branch :v:

**Comment by ahrtr** [comment]
@ivanvc could you please submit a new PR? The PR is automatically closed after I renamed the master to main.

**Pull request closed without merging**
